### PR TITLE
Mina/oauth

### PR DESCRIPTION
**Pull request by huyqbui**
Implemented Github Oauth Login via Passport. 

In backend, Passport will use a Github Strategy to retrieve github access token, and user profile. 

In electron.jsx (our MAIN PROCESS), we create a new BrowserWindow that takes the user to github authorization url.
Once user allows our request, github will grab their userprofile and we will log them in our app.

Upon successful login, the github BrowserWindow will automatically close, and user will be able to access OpenFolder Component.

Logout has not been linked with Github yet, so will need to work on that at a later point.

### CODE REVIEW

File: public/electron.jsx [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+
+const { app, BrowserWindow, ipcMain, dialog, webContents, session} = require('electron');
 const path = require('path');
 const fs = require('fs');
 const np = require('node-pty');
 const os = require('os');
 const server = require('../server/server.js');
 // react developer tools for electron in dev mode 
 const { default: installExtension, REACT_DEVELOPER_TOOLS } = require('electron-devtools-installer');
+const { ipcRenderer } = require('electron');
```
Comment (Ruzeb) [inline public/electron.jsx:12]: do we need ipcRenderer in this file anymore? if not let's remove it

File: public/electron.jsx [hunk 2]
```diff
@@ -132,40 +137,95 @@ ipcMain.on('ExportFileModal.mkdir', (e, folderPath) => {
...
+			githubWindow.close();
+			// app.webContents.send('final-url', 'reached final localhost url');
+		}
+	})
+        	
+})
+
+
+
+
```
Comment (Ruzeb) [inline public/electron.jsx:229]: revert/modify to 2 spaced tabs please

File: src/pages/ProjectLoader/ProjectLoader.jsx [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { GlobalContext } from '../../context/reducers/globalReducer';
 import {setGuest} from '../../context/actions/globalActions'
 import OpenFolder from '../../components/OpenFolder/OpenFolderButton.jsx';
 import { Button, TextField } from '@material-ui/core';
+const { BrowserWindow, ipcRenderer } = require('electron');
+// const remote = require('@electron/remote/main')
```
Comment (Ruzeb) [inline src/pages/ProjectLoader/ProjectLoader.jsx:8]: remove if not needed

File: public/electron.jsx (hunk 2 above)
Comment (Ruzeb) [inline public/electron.jsx:229]: revert/modify to 2 spaced tabs please

File: src/pages/ProjectLoader/ProjectLoader.jsx (hunk 3 above)
Comment (Ruzeb) [inline src/pages/ProjectLoader/ProjectLoader.jsx:8]: remove if not needed

File: public/electron.jsx (hunk 1 above)
Comment (Ruzeb) [inline public/electron.jsx:12]: do we need ipcRenderer in this file anymore? if not let's remove it

**Review by Ruzeb — CHANGES_REQUESTED** [review]

File: public/electron.jsx (hunk 1 above)
Comment (huyqbui) [reply public/electron.jsx:12]: agreed, let's remove it
```suggestion
```

File: src/pages/ProjectLoader/ProjectLoader.jsx (hunk 3 above)
Comment (huyqbui) [reply src/pages/ProjectLoader/ProjectLoader.jsx:8]:
```suggestion
```

**Pull request merged**